File: Signiture-Backend/kotlin/delivery/db/src/main/java/pcy/study/db/store/Store.java
package pcy.study.db.store;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.*;
import pcy.study.db.BaseTimeEntity;
import pcy.study.db.store.enums.StoreCategory;
import pcy.study.db.store.enums.StoreStatus;

import java.math.BigDecimal;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@ToString
@Entity
public class Store extends BaseTimeEntity {

    @Column(length = 100, nullable = false)
    private String name;

    @Column(length = 150, nullable = false)
    private String address;

    @Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @Enumerated(EnumType.STRING)
    private StoreStatus status;

    @Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @Enumerated(EnumType.STRING)
    private StoreCategory category;

    private double star;

    @Column(length = 200, nullable = false)
    private String thumbnailUrl;

    @Column(precision = 11, scale = 4, nullable = false)
    private BigDecimal minimumAmount;

    @Column(precision = 11, scale = 4, nullable = false)
    private BigDecimal minimumDeliveryAmount;

    @Column(length = 20)
    private String phoneNumber;

    @Builder
    private Store(String name, String address, StoreCategory category, String thumbnailUrl, BigDecimal minimumAmount, BigDecimal minimumDeliveryAmount, String phoneNumber) {
        this.name = name;
        this.address = address;
        this.category = category;
        this.thumbnailUrl = thumbnailUrl;
        this.minimumAmount = minimumAmount;
        this.minimumDeliveryAmount = minimumDeliveryAmount;
        this.phoneNumber = phoneNumber;
        this.status = StoreStatus.REGISTERED;
    }
}

File: Signiture-Backend/kotlin/delivery/db/src/main/kotlin/pcy/study/db/store/Store.kt
package pcy.study.db.store

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import pcy.study.db.store.enums.StoreCategory
import pcy.study.db.store.enums.StoreStatus
import java.math.BigDecimal
import java.time.LocalDateTime

@Entity
data class Store @JvmOverloads constructor(

    @field:Id
    @field:GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null,


    @field:Column(length = 100, nullable = false)
    var name: String,

    @field:Column(length = 150, nullable = false)
    val address: String,

    @field:Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @field:Enumerated(EnumType.STRING)
    var category: StoreCategory,

    var star: Double? = 0.0,

    @field:Column(length = 200, nullable = false)
    var thumbnailUrl: String,

    @field:Column(precision = 11, scale = 4, nullable = false)
    var minimumAmount: BigDecimal,

    @field:Column(precision = 11, scale = 4, nullable = false)
    var minimumDeliveryAmount: BigDecimal,

    @field:Column(length = 20)
    var phoneNumber: String,

    @field:Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @field:Enumerated(EnumType.STRING)
    var status: StoreStatus? = StoreStatus.REGISTERED,

    @field:CreationTimestamp
    @field:Column(updatable = false)
    private var createdAt: LocalDateTime? = null,

    @field:UpdateTimestamp
    private var updatedAt: LocalDateTime? = null,
)